### DISCUSSION TITLE
Makefile error during installation

### DISCUSSION THREAD

**Discussion by cartilage-ftw** [iraf-community/iraf#138] [Q&A]
I've been trying to install IRAF [2.16.1+2018.11.01](https://github.com/iraf-community/iraf/releases/tag/v2.16.1+2018.11.01) using [this documentation here](https://iraf-community.github.io/install.html).

I'm using Manjaro KDE Plasma (64 bit architecture). I have the following dependencies installed.
```bash
flex                                                            2.6.4-3   core 
    A tool for generating text-scanning programs
python-pycurl                                              7.43.0.6-1  extra 
    A Python 3.x interface to libcurl
lib32-curl                                                 7.73.0-1    multilib 
    An URL retrieval utility and library (32-bit)
curl                                                       7.75.0-1    core 
    An URL retrieval utility and library
lib32-expat                                                  2.2.10-1  multilib 
    An XML Parser library
expat                                                        2.2.10-2  core 
    An XML parser library
lib32-readline                                               8.0.0-1   multilib 
    GNU readline library (32-bit)
readline                                                     8.1.0-2   core 
    GNU readline library
```
apart from the ```gcc```, ```perl```, and ```make``` packages.

Please let me know if I'm missing something -- package names in Arch are different from those for Ubuntu I feel. I'm attaching my [build.log](https://github.com/iraf-community/iraf/files/6289246/build.log).

The bottom of the file reads

```bash
  197 | }
      | ^
getproc.c: In function ‘uid_executing’:
getproc.c:24:29: warning: ‘%s’ directive writing up to 255 bytes into a region of size 250 [-Wformat-overflow=]
   24 |      sprintf (fname, "/proc/%s", direntp->d_name);
      |                             ^~
getproc.c:24:6: note: ‘sprintf’ output between 7 and 262 bytes into a destination of size 256
   24 |      sprintf (fname, "/proc/%s", direntp->d_name);
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:19: sysgen] Error 1
```

And of course, this means that when I run the tests, I get
```bash
IRAF shell ecl.e not found
stat: cannot statx '/home/aayush/Documents/iraf-2.16.1-2018.11.01/bin.linux64/ecl.e': No such file or directory
```

**Comment by olebole** [comment]
You should search for error messages in the log file, and they are some already at the very beginning:

```
+ gcc -c -g -Wall -O2 -I/home/aayush/Documents/iraf-2.16.1-2018.11.01/include -Wall ../as/zsvjmp.S gmttolst.c irafpath.c prwait.c zalloc.c zawset.c zcall.c zdojmp.c zfacss.c zfaloc.c zfchdr.c zfdele.c zfgcwd.c zfinfo.c zfiobf.c zfioks.c zfiolp.c zfiomt.c zfiond.c zfiopl.c zfiopr.c zfiosf.c zfiotx.c zfioty.c zflink.c zfmkcp.c zfmkdr.c zfnbrk.c zfpath.c zfpoll.c zfprot.c zfrmdr.c zfrnam.c zfsubd.c zfunc.c zfutim.c zfxdir.c zgcmdl.c zghost.c zglobl.c zgmtco.c zgtenv.c zgtime.c zgtpid.c zintpr.c zlocpr.c zlocva.c zmain.c zmaloc.c zmfree.c zopdir.c zopdpr.c zoscmd.c zpanic.c zraloc.c zshlib.c zttyio.c zwmsec.c zxwhen.c zzdbg.c zzepro.c zzexit.c zzpstr.c zzsetk.c zzstrt.c alloc.c getproc.c
In file included from irafpath.c:11:
/home/aayush/Documents/iraf-2.16.1-2018.11.01/include/iraf.h:1:1: error: expected identifier or ‘(’ before ‘.’ token
    1 | ../unix/hlib/libc/iraf.h
      | ^
In file included from prwait.c:11:
/home/aayush/Documents/iraf-2.16.1-2018.11.01/include/iraf.h:1:1: error: expected identifier or ‘(’ before ‘.’ token
    1 | ../unix/hlib/libc/iraf.h
      | ^
prwait.c:31:11: error: ‘MAXPROCS’ undeclared here (not in a function)
   31 | } prtable[MAXPROCS];
      |           ^~~~~~~~
```
I would guess that there went something wrong with the creation of symlinks when you had run `./install` -- they were created as a text file instead of a symlink. This is quite mystic, and I would guess that you use somehow a non-standard file system that does not support symlinks?

**Reply by cartilage-ftw** [reply]
I'm quite a novice with makefile logs. I'm not sure what to do at this point. How would one go on about debugging this?

**Reply by olebole** [reply]
I would really recommend to **not** start with IRAF, but instead use modern software, like [Astropy](https://astropy.org) and its [affiliated packages](https://www.astropy.org/affiliated/index.html). IRAF is deprecated since tens of years and has no future. Keeping it running for those who cannot switch is OK, but starting anything new with it is really a bad idea and leads you into a dead end.

You should check whether the `include/iraf.h` is a symlink or a normal file. If it is a normal file, this is wrong, Then you should try to find out why it was created as a file instead of a link. Since they come from the package file (tar.gz or zip), probably something went wrong during unpacking. So, you should check this step and the command line options you used there.
